modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteUnionAll.java

```diff
@@ -132,6 +134,30 @@ public IgniteUnionAll(RelInput input) {
             Commons.transform(inputTraits, t -> t.replace(RelCollations.EMPTY))));
     }
 
+    /** {@inheritDoc} */
+    @Override public List<Pair<RelTraitSet, List<RelTraitSet>>> passThroughCorrelation(RelTraitSet nodeTraits,
+        List<RelTraitSet> inTraits) {
+
+        if (TraitUtils.correlation(nodeTraits).correlated())
```

Comment:
Why don't we just pass all correlation like we did it in IgniteNestedLoopJoin?

Reply:
fixed